### PR TITLE
Implement Server Client Exception 

### PR DESCRIPTION
**Pull request by Thumimku**
## Purpose
Implement server and client exceptions.

## Approach
- Define proper error code in Constants.
- Throw client exception/server exception 
- Generate client /server exception if no exception throws.(handle server/client exception)

### CODE REVIEW

File: components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/RemoteFetchConstants.java [hunk 1]
```diff
@@ -64,4 +64,60 @@ private RemoteFetchTableConstants() {
...
+                "Remote Fetch Configuration with resource ID: %s does not exists."),
+        ERROR_CODE_RF_CONFIG_ADD_REQUEST_INVALID("RFE-60003",
+                "Remote Fetch Configuration add request validation failed. %s"),
+        ERROR_CODE_RF_CONFIG_GET_REQUEST_INVALID("RFE-60004",
+                "Remote Fetch Configuration get request validation failed. %s"),
+        ERROR_CODE_RF_CONFIG_DELETE_REQUEST_INVALID("RFE-60005",
+                "Remote Fetch Configuration delete request validation failed. %s"),
+        ERROR_CODE_RF_CONFIG_UPDATE_REQUEST_INVALID("RFE-60006",
+                "Remote Fetch Configuration update request validation failed. %s");
+
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/RemoteFetchConstants.java:96]: Remove extra empty line

File: components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/util/RemoteFetchConfigurationUtils.java [hunk 2]
```diff
@@ -122,4 +127,102 @@ public static int getMaximumItemPerPage() {
...
+    /**
+     * This method is used to generate RemoteFetchClientException from RemoteFetchConstants.ErrorMessage
+     * when no exception is thrown.
+     * @param error RemoteFetchConstants.ErrorMessage
+     * @param data data to replace if message needs to be replaced.
+     * @return RemoteFetchClientException
+     */
+    public static RemoteFetchClientException handleClientException(RemoteFetchConstants.ErrorMessage error,
+                                                                   List<String> data) {
+
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/util/RemoteFetchConfigurationUtils.java:156]: Remove extra empty line

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/exceptions/RemoteFetchServerException.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ */
+
+package org.wso2.carbon.identity.remotefetch.common.exceptions;
+
+/**
+ * Exception class for Remote Fetch Configuration server related exceptions.
+ */
+public class RemoteFetchServerException extends RemoteFetchCoreException {
+
+    public RemoteFetchServerException(String message) {
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/exceptions/RemoteFetchServerException.java:26]: Formatting. Newline

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/exceptions/RemoteFetchCoreException.java [hunk 4]
```diff
@@ -33,4 +35,39 @@ public RemoteFetchCoreException(String message, Throwable cause) {
         super(message, cause);
     }
 
+    /**
+     * Constructs a new exception with the specified error code and cause.
+     *
+     * @param errorCode Error code
+     * @param message   Detailed message
+     */
+    public RemoteFetchCoreException(String errorCode, String message) {
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/exceptions/RemoteFetchCoreException.java:44]: What is the expectation of the RemoteFetchCoreException and RemoteFetchServerException. When should these be used?

Comment (Thumimku) [reply components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/exceptions/RemoteFetchCoreException.java:44]: RemoteFetchServerException is a child class of RemoteFetchCoreException. RemoteFetchServerException is used to point out a server error when processing user requests (OSGi service calls). 
RemoteFetchCoreException is used elsewhere to point out other exceptions.

File: components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/RemoteFetchConstants.java (hunk 1 above)
Comment (Thumimku) [reply components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/RemoteFetchConstants.java:96]: Resolved

File: components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/util/RemoteFetchConfigurationUtils.java (hunk 2 above)
Comment (Thumimku) [reply components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/util/RemoteFetchConfigurationUtils.java:155]: Resolved

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/exceptions/RemoteFetchServerException.java (hunk 3 above)
Comment (Thumimku) [reply components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/exceptions/RemoteFetchServerException.java:26]: Resolved

Comment (Thumimku) [reply components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/exceptions/RemoteFetchServerException.java:26]: Resolved

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/exceptions/RemoteFetchCoreException.java (hunk 4 above)
Comment (Thumimku) [reply components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/exceptions/RemoteFetchCoreException.java:44]: This particular method is to use to throw a new RemoteFetchServerException object(after handle server exception using Utils method.) with reference to its parent RemoteFetchCoreException.

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchComponent.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+/**
+ * Interface to implement to common gunctionalities of the component.
+ */
+public interface RemoteFetchComponent {
+
+    /**
+     * Returns a string of the unique identifier of the component.
+     *
+     * @return
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchComponent.java:33]: the return parameter is missing.

**Pull request merged**